From 35f2da263554d781a87d3257aa3b01185ff588f2 Mon Sep 17 00:00:00 2001
From: mariaum <me@mariaum.com>
Date: Fri, 22 Sep 2023 12:14:07 -0300
Subject: [PATCH] Fix NPE with latest Bungee changes.

Netty fully clears the pipeline after closing down the channel. Due to this getEncodeProtocol() will throw a NPE after the channel has been closed. The isObsolete() call should remediate this.

diff --git a/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java b/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java
index 5f4b4b672..2d4747dd8 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java
@@ -135,7 +135,7 @@ public class UpstreamBridge extends PacketHandler
     @Override
     public void handle(PacketWrapper packet) throws Exception
     {
-        if ( con.getServer() != null )
+        if ( con.getServer() != null && !con.getServer().isObsolete() )
         {
             EntityMap rewrite = con.getEntityRewrite();
             if ( rewrite != null && con.getServer().getCh().getEncodeProtocol() == Protocol.GAME )
-- 
2.40.1.windows.1

